auth/provider/types.go

```diff
@@ -0,0 +1,39 @@
+package provider
+
+import (
+	"strings"
+	"time"
+)
+
+type Profile struct {
+	ID        int       `json:"id"`
+	Login     string    `json:"login"`
+	Name      string    `json:"name"`
+	Email     string    `json:"email"`
+	TwoFactor bool      `json:"two_factor"`
+	CreatedAt time.Time `json:"created_at"`
+}
+
+type Provider interface {
+	GetProfile(accessToken string) (*Profile, error)
+}
+
+var supportedProviders = []string{
+	"github",
+	"gitlab",
+}
+
+func GetSupportedString() {
+	strings.Join(supportedProviders, ", ")
+}
+
+func IsSupported(name string) bool {
+	for _, sp := range supportedProviders {
+		if strings.ToLower(name) == strings.ToLower(sp) {
```

Comment:
Perhaps use `EqualFold` which is designed for this use-case: https://golang.org/pkg/strings/#EqualFold

Reply:
Changed